src/repr/scalar/strconv.rs

```diff
@@ -167,8 +307,46 @@ where
     format_nanos(buf, ts.timestamp_subsec_nanos());
 }
 
-pub fn parse_interval(_s: &str) -> Result<Interval, failure::Error> {
-    bail!("parsing the interval text format is not yet supported")
+/// parse
+///
+/// ```text
+/// <unquoted interval string> ::=
+///   [ <sign> ] { <year-month literal> | <day-time literal> }
+/// <year-month literal> ::=
+///     <years value> [ <minus sign> <months value> ]
+///   | <months value>
+/// <day-time literal> ::=
+///     <day-time interval>
+///   | <time interval>
+/// <day-time interval> ::=
+///   <days value> [ <space> <hours value> [ <colon> <minutes value>
+///       [ <colon> <seconds value> ] ] ]
+/// <time interval> ::=
+///     <hours value> [ <colon> <minutes value> [ <colon> <seconds value> ] ]
+///   | <minutes value> [ <colon> <seconds value> ]
+///   | <seconds value>
+/// ```
+pub fn parse_interval(s: &str) -> Result<Interval, failure::Error> {
+    parse_interval_w_disambiguator(s, DateTimeField::Second)
+}
+
+/// Parse an interval string, using a specific sql_parser::ast::DateTimeField
+/// to identify ambiguous elements. For more information about this operation,
+/// see the doucmentation on ParsedDateTime::build_parsed_datetime_interval.
+pub fn parse_interval_w_disambiguator(
```

Comment:
Can this be private?

Reply:
Converting `IntervalValue` to `Interval` requires this function, which will now be happening [in `sql/query.rs`](https://github.com/MaterializeInc/materialize/pull/2026/files#diff-5219e3cc04c3438197190bf3898baa16R2871), so looks like it needs to be public.

Reply:
Hmm, ok, we can clean up later.